### PR TITLE
Judge pull request

### PR DESCRIPTION
**Pull request by xKoem**
Created Judge microservice with tests.

### CODE REVIEW

**Comment by boczkas** [comment]
Build 132 Failure

File: Judge/src/main/java/pl/jakubowskiprzemyslaw/tajgertim/service/JudgeQueueListener.java [hunk 1]
```diff
@@ -23,18 +23,22 @@ public JudgeQueueListener(QueueService queueService, LoggerService logger) {
...
 
-        RoundStatus roundStatus = RoundStatus.GAME_END;
-
-        if (shootResult.equals(ShootResult.HIT)) {
-            roundStatus = RoundStatus.SAME_PLAYER;
-        }
-        else if (shootResult.equals(ShootResult.MISS)) {
-            roundStatus = RoundStatus.NEXT_PLAYER;
+        RoundStatus roundStatus;
+        switch (shootResult) {
```
Comment (makoJohn) [inline Judge/src/main/java/pl/jakubowskiprzemyslaw/tajgertim/service/JudgeQueueListener.java]: playerShootResult.getShootResult()

File: Judge/src/test/java/pl/jakubowskiprzemyslaw/tajgertim/service/JudgeQueueListenerTest.java [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+public class JudgeQueueListenerTest {
+
+    private Player player;
+    private QueueService queueService;
+    private JudgeQueueListener judgeQueueListener;
+
+    @BeforeMethod
+    public void setUp() {
+        player = new Player("", "");
+        queueService = mock(QueueService.class);
```
Comment (makoJohn) [inline Judge/src/test/java/pl/jakubowskiprzemyslaw/tajgertim/service/JudgeQueueListenerTest.java]: you can use @Mock annotation as well

File: Judge/src/test/java/pl/jakubowskiprzemyslaw/tajgertim/service/JudgeQueueListenerTest.java [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+        judgeQueueListener = new JudgeQueueListener(queueService, loggerService);
+    }
+
+    public void shootResult_HIT_sendToQueueRoundStatus_SAME_PLAYER() {
+        PlayerShootResult playerShootResult = new PlayerShootResult(player,ShootResult.HIT);
+
+        judgeQueueListener.listenOnJudgePlayerShootResultQueue(playerShootResult);
+
+        NextRoundStatus nextRoundStatus = new NextRoundStatus(RoundStatus.SAME_PLAYER);
```
Comment (makoJohn) [inline Judge/src/test/java/pl/jakubowskiprzemyslaw/tajgertim/service/JudgeQueueListenerTest.java]: inline this variable

**Comment by xKoem** [comment]
Closing because some errors occurred during rebase

File: Judge/src/main/java/pl/jakubowskiprzemyslaw/tajgertim/service/JudgeQueueListener.java (hunk 1 above)
Comment (boczkas) [reply Judge/src/main/java/pl/jakubowskiprzemyslaw/tajgertim/service/JudgeQueueListener.java]: done according to articles like:
https://blogs.msdn.microsoft.com/cellfish/2009/08/19/object-calisthenics-rule-4-use-only-one-dot-per-line/

File: Judge/src/test/java/pl/jakubowskiprzemyslaw/tajgertim/service/JudgeQueueListenerTest.java (hunk 3 above)
Comment (boczkas) [reply Judge/src/test/java/pl/jakubowskiprzemyslaw/tajgertim/service/JudgeQueueListenerTest.java]: done according to articles like:
https://blogs.msdn.microsoft.com/cellfish/2009/08/19/object-calisthenics-rule-4-use-only-one-dot-per-line/

File: Judge/src/test/java/pl/jakubowskiprzemyslaw/tajgertim/service/JudgeQueueListenerTest.java (hunk 2 above)
Comment (boczkas) [reply Judge/src/test/java/pl/jakubowskiprzemyslaw/tajgertim/service/JudgeQueueListenerTest.java]: we will consider that

**Pull request closed without merging**